src/Orleans.TestingHost/StandaloneSiloHost.cs

```diff
@@ -0,0 +1,134 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans.TestingHost
+{
+    /// <summary>
+    /// The entry point for standalone silo processes. See <see cref="StandaloneSiloHandle" />.
+    /// </summary>
+    public static class StandaloneSiloHost
+    {
+        public const string SiloAddressLog = "#### SILO ";
+        public const string GatewayAddressLog = "#### GATEWAY ";
+        public const string StartedLog = "#### STARTED";
+        public const string ShutdownCommand = "#### SHUTDOWN";
+
+        public static async Task Main(string[] args)
+        {
+            if (args.Length < 2)
+            {
+                Console.Error.WriteLine("Expected JSON-serialized configuration to be provided as an argument");
+            }
+
+            var monitorProcessId = int.Parse(args[0], NumberStyles.Integer, CultureInfo.InvariantCulture);
+            var serializedConfiguration = args[1];
+            var configuration = TestClusterHostFactory.DeserializeConfiguration(serializedConfiguration);
+            var name = configuration["SiloName"];
+            var host = TestClusterHostFactory.CreateSiloHost(name, configuration);
+            var cts = new CancellationTokenSource();
+            Console.CancelKeyPress += (sender, eventArgs) => cts.Cancel();
+
+            ListenForShutdownCommand(cts);
+            MonitorParentProcess(monitorProcessId);
+
+            await host.StartAsync(cts.Token);
+
+            // This is a special marker line.
+            var localSiloDetails = (ILocalSiloDetails)host.Services.GetService(typeof(ILocalSiloDetails));
+            Console.WriteLine($"{SiloAddressLog}{localSiloDetails.SiloAddress.ToParsableString()}");
+            Console.WriteLine($"{GatewayAddressLog}{localSiloDetails.GatewayAddress.ToParsableString()}");
+            Console.WriteLine(StartedLog);
+
+            await cts.Token.WhenCancelled();
+
+            await host.StopAsync(CancellationToken.None);
+        }
+
+        private static void MonitorParentProcess(int monitorProcessId)
+        {
+            if (monitorProcessId > 0)
+            {
+                Console.WriteLine($"Monitoring parent process {monitorProcessId}");
+                Process.GetProcessById(monitorProcessId).Exited += (o, a) =>
+                {
+                    Console.Error.WriteLine($"Parent process {monitorProcessId} has exited");
+                    Environment.Exit(0);
+                };
+
+                _ = Task.Factory.StartNew(async _ =>
+                {
+                    try
+                    {
+                        while (true)
+                        {
+                            await Task.Delay(5000);
+                            if (!Process.GetProcesses().Any(p => p.Id == monitorProcessId))
+                            {
+                                Console.Error.WriteLine($"Parent process {monitorProcessId} has exited");
```

Comment:
I suppose it's fine since the parent process died already, but just double-checking whether you want to attempt a graceful shutdown? Maybe for some users that's helpful even though not ideal to rely on it?

Reply:
I considered it, but I think that fast killing is more appropriate here. We want to avoid zombies and we don't want to annoy developers when they're rebuilding their app but some old process is locking their files